pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2593,7 +2593,8 @@ public void receiveParameterStatus() throws IOException, SQLException {
           value), PSQLState.CONNECTION_FAILURE);
     }
 
-    if (name.equals("DateStyle") && !value.startsWith("ISO,")) {
+    if (name.equals("DateStyle") && !value.startsWith("ISO")
+        && !value.toUpperCase().startsWith("ISO")) {
```

Comment:
It looks redundant to check startsWith() first and toUpperCase().startsWith() later.

Reply:
The first one is an optimization